src/main/java/org/jetbrains/research/refactorinsight/services/WindowService.java

```diff
@@ -35,15 +37,39 @@ public static WindowService getInstance(@NotNull Project project) {
    * @param state true for selected, false for unselected
    */
   public void setSelected(@NotNull MainVcsLogUi ui, boolean state) {
-    GitWindow gitWindow = gitInfo.computeIfAbsent(ui.getTable(), table -> {
-      Disposer.register(ui, () -> gitInfo.remove(ui.getTable()));
-      return new GitWindow(project, ui);
-    });
+    GitWindow gitWindow = gitInfo.get(ui.getTable());
```

Comment:
Could you please clarify this change?

Reply:
This has been moved to the update function. This ensures labels can be viewed before the refactorings view is opened!